### PR TITLE
[msal-extensions] Extensions #3

### PR DESCRIPTION
**Pull request by sangonzal**
- Add persistence on Linux, Windows, and Mac
- Add persistence plugin, which can be set on MSAL

### CODE REVIEW

File: extensions/src/persistence/FilePersistence.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        await closePromise(await openPromise(this.filePath, "a"))
+    }
+
+    private async createFileDirectory(): Promise<void> {
+        try {
+            const mkdirPromise = promisify(mkdir);
+            await mkdirPromise(dirname(this.filePath), {recursive: true});
+        } catch (err) {
+            if (err.code == Constants.EEXIST_ERROR) {
+                console.log(`Directory ${dirname(this.filePath)} " already exists"`);
```
Comment (sangonzal) [inline extensions/src/persistence/FilePersistence.ts:110]: every console.log statement will be refactored to use the msal-common logger in an upcoming PR

**Comment by pkanher617** [comment]
Same comment about unit tests

File: extensions/src/persistence/PersistenceCachePlugin.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                console.log("Reload necessary.  Last sync time: " + this.lastSync);
+                this.crossPlatformLock = new CrossPlatformLock(this.lockFilePath, this.lockOptions);
+                await this.crossPlatformLock.lock();
+
+                this.currentCache = await this.persistence.load();
+                this.lastSync = new Date().getTime();
+                console.log("Last sync time updated to: ", this.lastSync);
+            } finally {
+                await this.crossPlatformLock.unlock();
+                delete this.crossPlatformLock;
```
Comment (jasonnutter) [inline extensions/src/persistence/PersistenceCachePlugin.ts:59]: I don't think this is the best use of `delete`. Setting to `null` would be better, as it wouldn't actually remove the property.

Comment (sangonzal) [reply extensions/src/persistence/PersistenceCachePlugin.ts:59]: Makes sense. Thinking about it futher, I think the lock object can be reused, so completely removed this.

File: extensions/msal-node-extensions/src/lock/CrossPlatformLock.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export class CrossPlatformLock {
         try {
             // delete lock file
```
Comment (sameerag) [inline extensions/msal-node-extensions/src/lock/CrossPlatformLock.ts:58]: Should the comment change? As you are `unlinking` and not `deleting`?

Comment (sangonzal) [reply extensions/msal-node-extensions/src/lock/CrossPlatformLock.ts:58]: We are deleting the file. https://nodejs.org/api/fs.html#fs_fs_unlink_path_callback

**Pull request merged**